build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -149,71 +225,138 @@ function getFiles(testType) {
 }
 
 /**
- * Adds reporters to the default karma spec per test settings.
- * Overrides default reporters for verbose settings.
+ * Recomputes the set of files for Karma to load.
+ * - Non-JS files are included as they are.
+ * - JS files are imported into a unified file for esbuild to transform at once.
+ * - The list of all non-JS files plus the unified JS file is returned.
  * @param {!RuntimeTestConfig} config
  */
-function updateReporters(config) {
-  if (
-    (argv.testnames || argv.local_changes || argv.files || argv.verbose) &&
-    !isCiBuild()
-  ) {
-    config.reporters = ['mocha'];
-  }
+function updateFiles(config) {
+  const fileGlobs = getFiles(config.testType);
 
-  if (argv.coverage) {
-    config.reporters.push('coverage-istanbul');
+  const isJsGlob = (glob) => typeof glob === 'string' && glob.endsWith('.js');
```

Comment:
This is dependent on us using `.js` as the only ending token, but any change to use more advanced globby features could break that. Eg, future support for TS and changing the glob to `.{js,ts}`. I'd rather this be explicit about fixing esbuild generation by looking for keys with values of `['esbuild', ...]`.

Reply:
Maintaining two lists, one for which files to include in the Karma server, and one for which files to run transforms on was unnecessary, because we transform every JS file with `esbuild` (that was the original intent). However, I see your point that not every JS glob must end with JS. One way to deal with this is to expand all the globs before processing. Will do this.

Reply:
Aha, I now see why we need two lists. There are JS files in `examples/` that shouldn't be transformed. I stand corrected. I've modified the logic to expand globs, but still automatically detect which files we want to transform with `esbuild` and which ones we don't. Lemme know what you think.